src/main/scala/org/phenoscape/kb/Similarity.scala

```diff
@@ -233,6 +237,53 @@ object Similarity {
       } yield JaccardScore(Set(left, right), intersectionCount.toDouble / unionCount.toDouble)).toSeq
     }
 
+  // Output a boolean matrix as CSV
+  def matrixRendererFromMapOfMaps(mapOfMaps: Map[Any, Map[Any, Boolean]]) = {
+
+    val sortedKeys = mapOfMaps.keys.toList.map(_.toString).sorted
```

Comment:
Don't convert all the keys to strings here. If you test with keys that are not strings, I think you will find this is a problem. You can sort using `.toString` using `.sortBy`.